File: src/Calculator/Controls/OverflowTextBlockAutomationPeer.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#include "pch.h"
#include "Controls/OverflowTextBlock.h"
#include "OverflowTextBlockAutomationPeer.h"

using namespace Windows::UI::Xaml::Automation::Peers;
using namespace Windows::Foundation::Collections;

namespace CalculatorApp::Controls
{
    OverflowTextBlockAutomationPeer::OverflowTextBlockAutomationPeer(OverflowTextBlock^ owner) :
        FrameworkElementAutomationPeer(owner)
    { }

    AutomationControlType OverflowTextBlockAutomationPeer::GetAutomationControlTypeCore()
    {
        return AutomationControlType::Text;
    }

    IVector<AutomationPeer^>^ OverflowTextBlockAutomationPeer::GetChildrenCore()
    {
        return nullptr;
    }
}

File: src/Calculator/Controls/OverflowTextBlockAutomationPeer.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;

using Windows.UI.Xaml.Automation.Peers;

namespace CalculatorApp
{
    namespace Controls
    {
        public sealed class OverflowTextBlockAutomationPeer : Windows.UI.Xaml.Automation.Peers.FrameworkElementAutomationPeer
        {
            public OverflowTextBlockAutomationPeer(OverflowTextBlock owner) : base(owner)
            {
            }

            protected override AutomationControlType GetAutomationControlTypeCore()
            {
                return AutomationControlType.Text;
            }

            protected override IList<AutomationPeer> GetChildrenCore()
            {
                return null;
            }
        }
    }
}
